pkg/registry/rbac/rest/storage_rbac.go

```diff
@@ -165,7 +166,30 @@ func isConflictOrServiceUnavailable(err error) bool {
 	return errors.IsConflict(err) || errors.IsServiceUnavailable(err)
 }
 
-func retryOnConflictOrServiceUnavailable(backoff wait.Backoff, fn func() error) error {
+type serviceTracker struct {
+	available atomic.Value
+}
+
+func newServiceTracker() *serviceTracker {
+	t := serviceTracker{}
+	t.available.Store(false)
+	return &t
+}
+
+// NOTE: we know that once the apiserver service (for one API resource) is
+// ready, it won't become not-ready again
+func (t *serviceTracker) markAvailable() {
+	t.available.Store(true)
+}
+
+func (t *serviceTracker) retryOnConflictOrServiceUnavailable(fn func() error) error {
+	backoff := retry.DefaultBackoff
+	if t.available.Load() == true {
+		return retry.RetryOnConflict(backoff, fn)
+	}
+	// We don't know if the service is available yet. Allow retrying on
+	// ServiceUnavailable error with longer time
+	backoff.Steps += 2
```

Comment:
6 steps from 10ms at a factor of 5 takes us to ~40 seconds retries for a single object:

```
10
50
250
1250
6250
31250
=====
39060
```

Our overall retry loop times out at 30 seconds. I'm concerned that this feature can delay startup so significantly (especially on a single server cluster where there should be zero contention/impact), and extending this retry fail the overall server startup if it takes too long

Reply:
discussed offline. We assumed the RBAC initialization times out at 30 seconds but it doesn't. Update the PR to fix the behavior.